tests/lib/rules/no-deprecated-destroyed-lifecycle.js

```diff
@@ -0,0 +1,268 @@
+/**
+ * @author Yosuke Ota
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-deprecated-destroyed-lifecycle')
+
+const RuleTester = require('eslint').RuleTester
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: require.resolve('vue-eslint-parser'),
+  parserOptions: { ecmaVersion: 2020, sourceType: 'module' }
+})
+ruleTester.run('no-deprecated-destroyed-lifecycle', rule, {
+  valid: [
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        unmounted () {},
+        beforeUnmount () {},
+      }
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        unmounted,
+        beforeUnmount,
+      }
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        beforeCreate,
+        created,
+        beforeMount,
+        mounted,
+        beforeUpdate,
+        updated,
+        activated,
+        deactivated,
+        beforeUnmount,
+        unmounted,
+        errorCaptured,
+        renderTracked,
+        renderTriggered,
+      }
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        beforeUnmount:beforeDestroy,
+        unmounted:destroyed,
+      }
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        ...beforeDestroy,
+        ...destroyed,
+      }
+      </script>
+      `
+    }
+  ],
+  invalid: [
```

Comment:
Should we add tests for computed properties? I'm not sure if this is a expected behavior but current implementation also warn for below code:
```js
export default {
  ["destroyed"]() {}
}
```

Reply:
Thank you for your review!

It is the expected behavior to report. But I should add a test case. In especially, I should test if `suggest` works.